tests/flow/test_optimizations_plan.py

```diff
@@ -284,3 +284,13 @@ def test16_test_splitting_cartesian_product_with_multiple_filters(self):
                     ['Roi']]
         resultset = graph.query(query).result_set
         self.env.assertEqual(resultset, expected)
+
+    def test17_test_multiple_branch_filter_cp_optimization(self):
+        query = """MATCH (p1), (p2), (p3), (p4) WHERE p1.val + p2.val = p3.val AND p3.val > 0 RETURN DISTINCT p3.name ORDER BY p3.name"""
```

Comment:
Why do you require `p3.val > 0` ?

Reply:
'Roi' value is 0, I don't want to include 'Roi' in the result set